packages/orchestrator-ui-components/src/components/WFOKeyValueTable/WFOKeyValueTable.tsx

```diff
@@ -0,0 +1,77 @@
+import React, { FC, ReactNode } from 'react';
+import { WFOClipboardCopy } from '../../icons/WFOClipboardCopy';
+import { EuiCopy } from '@elastic/eui';
+import { useOrchestratorTheme } from '../../hooks';
+import { getStyles } from './styles';
+
+export type WFOKeyValueTableDataType = {
+    key: string;
+    value: ReactNode;
+    plainTextValue?: string;
+};
+
+export type WFOKeyValueTableProps = {
+    keyValues: WFOKeyValueTableDataType[];
+    showCopyToClipboardIcon?: boolean;
+};
+
+export const WFOKeyValueTable: FC<WFOKeyValueTableProps> = ({
+    keyValues,
+    showCopyToClipboardIcon = false,
+}) => {
+    const { theme } = useOrchestratorTheme();
+    const {
+        keyColumnStyle,
+        valueColumnStyle,
+        copyColumnStyle,
+        clickable,
+        lightBackground,
+        darkBackground,
+    } = getStyles(theme);
+
+    return (
+        <table width="100%">
+            <tbody>
+                {keyValues.map(({ key, value, plainTextValue }, rowNumber) => {
```

Comment:
move this into a function, I do not like logic in the JSX

Reply:
It is split up in a key and cell component